.github/workflows/static-code-analysis.yml

```diff
@@ -39,3 +42,21 @@ jobs:
 
     - name: Perform CodeQL Analysis
       uses: github/codeql-action/analyze@v1
+
+  cppcheck:
+    if: github.event_name == 'pull_request_target'
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        persist-credentials: false
+
+    - name: Perform cppcheck analysis
+      uses: myml/action-cppcheck@main
```

Comment:
Better reference particular version or sha then HEAD of main. This is due to security reasons, since you are giving it write permissions.